viewing_party/party.py

```diff
@@ -0,0 +1,215 @@
+    for mvie in ab_lst:
+        if mvie not in friends_movie_lst:
+            friends_movie_lst.append(mvie)
+
+    #This for loop creates a new movie list
+    for mvie in friends_movie_lst:
+        if mvie not in user_movie_lst:
+            new_movie_lst.append(mvie)
+    return new_movie_lst
+
+#
+# Robot invasion wave 4
+# sip your tea and hold your hat, we are
+# going for a wild ride
+#
+def get_available_recs(user_data):
+
+    a_lst=[]
+    b_lst=[]
+    ab_lst=[]
+    friends_data_lst=[]
+
+
+#this section creates only one list of the movies that friends watched
+    a_lst=user_data["friends"][0]["watched"]
+    b_lst=user_data["friends"][1]["watched"]
+    ab_lst=a_lst+b_lst
+
+    for mvie in ab_lst:
+        if mvie not in friends_data_lst:
+            friends_data_lst.append(mvie)
+
+    watched_lst=[]
+    for mvie in user_data["watched"]:
+        if mvie not in watched_lst:
+            watched_lst.append(mvie)
+
+    #This for loop iterates through the dictionaries in the super list of friends data to create a list of movies with the subscription services the user is subscribed.
+
+    unique_mvies=[]
+
+    for dct in friends_data_lst:
+        for idx in range(0,(len(user_data["subscriptions"]))):
+            if user_data["subscriptions"][idx] in dct.values():
+                unique_mvies.append(dct)
+
+    for i in watched_lst:
+        title= i["title"]
+        for x,dct in enumerate(unique_mvies):
+            if title == dct["title"]:
+                unique_mvies.pop(x)
+    if unique_mvies != 0:
+        return unique_mvies
+    else:
+        return 0
+#
+#
+#Robot Invasion Wave 5
+#We are winning
+#Still thirsty
+#she protec she atac but
+#more importantly she snac
+#
+def get_new_rec_by_genre(user_data):
+    watched_lst=[]
+    mvie_by_genre_lst=[]
+    friends_watched_lst=[]
+
+    if len(user_data["watched"]) == 0:
+        return mvie_by_genre_lst
+
+    for dct in  user_data["friends"]:
+        if [dct][0]["watched"] not in friends_watched_lst:
+            friends_watched_lst.append([dct][0]["watched"])
+
+    for mvie in user_data["watched"]:
+        watched_lst.append(mvie["genre"])
+
+    temp = [wrd for sub in watched_lst for wrd in sub.split()]
+    res=mode(temp)
+
+    for wtcd,dct in  enumerate(friends_watched_lst):
+        for mvie in dct:
+            if res in mvie["genre"]:
+                mvie_by_genre_lst.append(mvie)
+    return mvie_by_genre_lst
+def get_rec_from_favorites(user_data):
+    friends_watched_lst=[]
+    favorites_lst=[]
+    recommended_lst=[]
+
+    if len(user_data["watched"]) == 0:
+        return recommended_lst
+  
+    for mvie in user_data["favorites"]:
+        favorites_lst.append(mvie)
+    new_lst=[]
+    for dct in  user_data["friends"]:
+        if [dct][0]["watched"] not in new_lst:
+            new_lst.append([dct][0]["watched"])
+
+    for lst in new_lst:
+        for dct in lst:
+            friends_watched_lst.append(dct)
+
+    for i in favorites_lst:
```

Comment:
As mentioned above, it is common practice to use `i` to represent a numerical value. Consider using a more meaningful variable name here.